Horizon/Horizon/Sources/Common/Domain/GetProgramsInteractor.swift

```diff
@@ -43,4 +87,17 @@ final class GetProgramsInteractor {
             }
             .eraseToAnyPublisher()
     }
+
+    private func fetchCourseProgression() -> AnyPublisher<[GetCoursesProgressionResponse.EnrollmentModel], Never> {
+        let userId = appEnvironment.currentSession?.userID ?? ""
+        let request = GetCoursesProgressionRequest(userId: userId)
+
+        return Future<[GetCoursesProgressionResponse.EnrollmentModel], Never> { [appEnvironment] promise in
+            appEnvironment.api.makeRequest(request) { response, _, _ in
+                let enrollments = response?.data?.user?.enrollments ?? []
+                promise(.success(enrollments))
+            }
+        }
+        .eraseToAnyPublisher()
+    }
```

Comment:
Please create a `GetCoursesProgressionUseCase` so we can use `ReactiveStore` instead of using `appEnvironment.api`.

Reply:
Yes, I have reviewed all the code that uses GraphQL, and I followed the same approach. However, as far as I know, when using ReactiveStore, we must save the data to Core Data first in order to call the .getEntities() function.

Reply:
Yes, we could do that here. Most of the graphQL code is considered legacy, we try to avoid directly making requests through the api entity in the `appEnvironment`. 

You could create a core data entity called `CDCourseProgression` and then create a use case for it.